lib/hdfs.js

```diff
@@ -79,18 +79,22 @@ libhdfs3.prototype.read = function(path, cb) {
     var stream = new Readable({ highWaterMark: 64 * 1024 });
     var eof = false;
     stream._read = function () {
-        file.read(function(err, data) {
-            if (err) {
-                return stream.emit('error', err);
-            }
-            if (eof) {
-                return stream.push(null);
-            }
-            if (data.bytesRead > 0) {
-                stream.push(data.data);
-            }
-            eof = data.eof;
-        });
+        if (eof) {
+            // If end of file, stop reading
+            return stream.push(null);
+        } else {
+            file.read(function(err, data) {
+                if (err) {
+                    return stream.emit('error', err);
```

Comment:
You can ditch all of the return statements inside of `stream._read()` since everything is wrapped in if/else/else if blocks but doesn't really matter.